exporter/opentelemetry-exporter-prometheus-remote-write/README.rst

```diff
@@ -1,27 +1,248 @@
+   **(Optional)**
+
+Example with all the configuration options:
+
+.. code:: python
+
+   exporter = PrometheusRemoteWriteMetricsExporter(
+       endpoint="http://localhost:9009/api/prom/push",
+       timeout=30,
+       basic_auth={
+           "username": "user",
+           "password": "pass123",
+       },
+       headers={
+           "X-Scope-Org-ID": "5",
+           "Authorization": "Bearer mytoken123",
+       },
+       proxies={
+           "http": "http://10.10.1.10:3000",
+           "https": "http://10.10.1.10:1080",
+       },
+       tls_config={
+           "cert_file": "path/to/file",
+           "key_file": "path/to/file",
+           "ca_file": "path_to_file",
+           "insecure_skip_verify": true, # for developing purposes
+       }
+   )
+
+Securing the Exporter
+---------------------
+
+Authentication
+~~~~~~~~~~~~~~
+
+The exporter provides two forms of authentication which are shown below.
+Users can add their own custom authentication by setting the appropriate
+values in the ``headers`` dictionary
+
+1. Basic Authentication Basic authentication sets a HTTP Authorization
+   header containing a base64 encoded username/password pair. See `RFC
+   7617`_ for more information. This
+
+.. code:: python
+
+   exporter = PrometheusRemoteWriteMetricsExporter(
+       basic_auth={"username": "base64user",  "password": "base64pass"}
+   )
+
+2. Bearer Token Authentication This custom configuration can be achieved
+   by passing in a custom ``header`` to the constructor. See `RFC 6750`_
+   for more information.
+
+.. code:: python
+
+   header = {
+       "Authorization": "Bearer mytoken123"
+   }
+
+TLS
+~~~
+
+Users can add TLS to the exporter's HTTP Client by providing certificate
+and key files in the ``tls_config`` parameter.
+
+Supported Aggregators
+---------------------
+
+-  Sum
+-  MinMaxSumCount
+-  Histogram
+-  LastValue
+-  ValueObserver
+
+Error Handling
+--------------
+
+In general, errors are raised by the calling function. The exception is
+for failed requests where any error status code is logged as a warning
+instead.
+
+This is because the exporter does not implement any retry logic as it
+sends cumulative metrics data. This means that data will be preserved
+even if some exports fail.
+
+For example, consider a situation where a user increments a Counter
+instrument 5 times and an export happens between each increment. If the
+exports happen like so:
 
 ::
 
-    pip install opentelemetry-exporter-prometheus-remote-write
+   SUCCESS FAIL FAIL SUCCESS SUCCESS
+   1       2    3    4       5
 
+Then the recieved data will be:
 
-.. _Prometheus: https://prometheus.io/
-.. _OpenTelemetry: https://github.com/open-telemetry/opentelemetry-python/
+::
 
+   1 4 5
 
-References
-----------
+The end result is the same since the aggregations are cumulative
+Contributing
+------------
+
+This exporter's datapath is as follows:
+
+|Exporter datapath| *Entites with ``*`` after their name are not actual
```

Comment:
So this diagram is great, but already exists in the design doc. I think READMEs here should be simple and not as technical, so it's fine to keep those in the design doc.

Reply:
alright, removed!